### ISSUE TITLE
Logout the Agent from everywhere

### ISSUE THREAD

**Issue by SaqibOctave** [amazon-connect/amazon-connect-streams#843]
Hi, 
I have embeded the streams in my angular project. i wrote the logout mechanism for my application using amazon connect streams. it works perfectly fine. when i hit the logout function it log me out from the CCP. 
Problem:
If i opened the application in different tabs. and i logout from one tab. i was unable to identify the logout event. I was not be able to identify any event when i logout from other tab so i was not be able to route to the login page. 
What i want :
I want that, Can we get any event trigger when the agent is logedout? as we get the event agent.onStateChange(). even if the agent change the state from any tab even if the agent change his state from the default CCP.

**Comment by oharez** [comment]
Second this, with some additions, similar (perhaps) use case. We need to capture the events from the client on a hosted iframe, it is mandatory with our system that we do so on the client side. The server side already receives contact traces and agent status. What we really need, is a way to either restrict logging in to the CCP within the connect console, or at least (per this post) be able to fire a backend to AWS command to change the status of the user/log them out if necessary etc. I believe this may be doable as is? But it would be really nice to have a property : where (domain/page etc) the "primary" CCP is being hosted, and the ability to limit that. It would be useful for those with hosted iframes I imagine. Or better yet, just split the property in connect console that allows access to the CCP, to two options. Allow CCP in Console, and allow external CCP.

**Comment by tannera926** [comment]
@oharez could you open a separate issue and go into more detail on exactly what you want to accomplish here - I'm not sure I fully understand what your goals are.

**Comment by tannera926** [comment]
@SaqibOctave have you tried subscribing to the `TERMINATED` event? Documentation is [here](https://github.com/amazon-connect/amazon-connect-streams/issues/863). If you are using custom CCP, you will need to manually fire the `TERMINATE` event when an agent hits the logout endpoint.

**Comment by vpogudina** [comment]
I think the provided link leads to some other github topic, most likely tannera926 meant to send [this link](https://github.com/amazon-connect/amazon-connect-streams/blob/master/Documentation.md#logging-out)

**Comment by SaqibOctave** [comment]
@vpogudina  @tannera926 
I am using this method to logout from one place what i want is when a consumer open my application in different browsers, If he/she logout from one browser it should logout from all the browsers

**Comment by vpogudina** [comment]
Thank you for bringing that to our attention, we will look into it and see if we can update the behavior for logout between tabs. However the browser context is independent and logging out from one browser will not affect the logged state in another browser immediately. You will get the logged out state synced after hourly check for refreshed cookie

**Comment by SaqibOctave** [comment]
Thanks @vpogudina  Waiting for your kind response if there is any work around please let me know

**Comment by oharez** [comment]
Waiting on this as well, same use case. We have call center managers who open dashboards and views in the AWS Connect domain, but are using their CCP in SN, inside an iframe, and we handle events to trigger workflows and such within the CCP client side. (obviously I don't encourage this, and we are working to recreate the tools they need in AWS to be in our CRM domain, and retrain...) We also have managers who don't really understand login/logout, and assume that if an agent walks away from their PC or closes browser etc that "offline" not-routable status/state == logged out. I try explaining that logged in is more a session thing than an agent state thing, but it would be nice to log them out. I put in a feature request for this and an API command to do so if this isn't possible. The problem on the client side CCP in an iframe is the after window close etc events never seem to work as well in practice as they might propose to. If for example, their CRM session ends, the AWS session persists, and their status remains offline. I have no way to fire a command from CRM backend that looks at agent status and says "hey this person is logged out of their CRM workspace (where they are required to take calls) but they're still logged in to AWS Connect and showing available "

**Comment by bretticus-mc** [comment]
Logging out from one browser and having it logged out from another browser is not currently supported. Please open a support ticket for a feature request and we can route it to the auth team.
